### ISSUE TITLE
Fatal error: exception CErrors.UserError(_, _)

### ISSUE THREAD

**Issue by xavierleroy** [coq/coq#12291]
#### Description of the problem

Batch compilation (`coqc`) of source files containing bullet-related errors prints the correct error message then reports a fatal error `exception CErrors.UserError(_, _)`

#### Coq Version

Coq 8.11.1 compiled with OCaml 4.09.0.

#### Repro case

```
$ cat bullet.v
Goal True /\ True.
Proof.
  split.
- tauto.
- tauto.
- tauto.
Qed.
$ coqc bullet.v
File "./bullet.v", line 6, characters 0-1:
Error: Wrong bullet -: No more subgoals.

Fatal error: exception CErrors.UserError(_, _)

$ cat notfinished.v
Goal False /\ True.
Proof.
  split.
- auto.
- auto.
Qed.
$ coqc notfinished.v
File "./notfinished.v", line 5, characters 0-1:
Error: Wrong bullet -: Current bullet - is not finished.

Fatal error: exception CErrors.UserError(_, _)
```

**Comment by ejgallego** [comment]
Thanks for the report @xavierleroy , this was fixed in `master` by #11498, will submit a fix for the 8.11 branch.

**Comment by ejgallego** [comment]
Closed by #12294
